app/services/multi_health_checker_summary.rb

```diff
@@ -0,0 +1,14 @@
+MultiHealthCheckerSummary = Struct.new(:statuses) do
```

Comment:
I don't think it's the nested namespaces that fool Rails autoloader, it's assigning to a constant.

When you run a
```ruby
class Foo
  def foobar
  end
end
```
twice, it'll update the existing class definition in memory.

If we wanted to make these autoload-friendly, we'd need to do:

```ruby
MultiHealthCheckerSummary = Struct.new(:statuses) unless defined?(MultiHealthCheckerSummary)
class MultiHealthCheckerSummary
  # etc
end
```

Anyways I'm not sure this is worth it just for development....

Reply:
Yep, agreed about not being worth it